base/complex.jl

```diff
@@ -107,7 +107,7 @@ abs(z::Complex)  = hypot(real(z), imag(z))
 abs2(z::Complex) = real(z)*real(z) + imag(z)*imag(z)
 inv(z::Complex)  = conj(z)/abs2(z)
 inv{T<:Integer}(z::Complex{T}) = inv(float(z))
-sign(z::Complex) = z/abs(z)
+sign(z::Complex) = z==zero(z)?zero(z):z/abs(z)
```

Comment:
I don't think full type stability can be achieved, there's no guarantee that `z/abs(z)` will be integer for `z::Complex{Int64}`. For example:
```javascript
julia> z = 1 + 1im; sign(z)
0.7071067811865475 + 0.7071067811865475im
```